File: leetcode/217/ContainDuplicate.py
from typing import List
# class Solution(object):
#     def containsDuplicate(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: bool
#         """
#         hash = {}
#         for i, num in enumerate(nums):
#             if nums[i] in hash:
#                 return True
#             else:
#                 hash[nums[i]] = True
#         return False


# class Solution:
#     def containsDuplicate(self, nums: List[int]) -> bool:
#         hash = {}
#         for i, num in enumerate(nums):
#             if num in hash:
#                 return True
#             else:
#                 hash[num] = True
#         return False

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:

        return len(set(nums)) != len(nums)


sl = Solution()
print(sl.containsDuplicate([1, 2, 3, 1]))

File: leetcode/217/ContainDuplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

class Solution
{
public:
    bool containsDuplicate(vector<int> &nums)
    {

        unordered_set<int> hash;

        for (int i = 0; i < nums.size(); i++)
        {
            if (hash.find(nums[i]) != hash.end())
                return true;
            else
                hash.insert(nums[i]);
        }
        return false;
    }
};

int main()
{
    Solution sl;

    vector<int> nums = {1, 2, 3, 1};
    bool result = sl.containsDuplicate(nums);

    cout << ' ' << sl.containsDuplicate(nums);
    cout << "\n";
    return 0;
}

// gcc -o ContainDuplicate ContainDuplicate.cpp - lstdc++